compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -702,7 +703,7 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
     /// properly. Calling the method twice with the same `NodeId` is fine though.
     #[instrument(level = "debug", skip(self), ret)]
     fn lower_node_id(&mut self, ast_node_id: NodeId) -> hir::HirId {
-        assert_ne!(ast_node_id, DUMMY_NODE_ID);
+        debug_assert_ne!(ast_node_id, DUMMY_NODE_ID);
```

Comment:
Hot (called for most nodes in code), but the assert is trivial.
This asserts looks pretty useful.